providers/redhat/check/check_cve.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/facebookincubator/nvdtools/providers/redhat/schema"
+	"github.com/facebookincubator/nvdtools/rpm"
+	"github.com/facebookincubator/nvdtools/wfn"
+)
+
+var NoCheckers = errors.New("no applicable checkers")
+
+func CVEChecker(cve *schema.CVE) (rpm.Checker, error) {
+	var chks []rpm.Checker
+
+	if archks, err := affectedReleaseCheckers(cve); err != nil {
+		return nil, fmt.Errorf("can't construct checkers for affected release: %v", err)
+	} else {
+		chks = archks
+	}
+
+	if pschks, err := packageStateCheckers(cve); err != nil {
+		return nil, fmt.Errorf("can't construct checkers for package state: %v", err)
+	} else {
+		chks = append(chks, pschks...)
+	}
+
+	if len(chks) == 0 {
+		return nil, NoCheckers
+	}
+
+	return &cveChecker{cve.Name, chks}, nil
+}
+
+func affectedReleaseCheckers(cve *schema.CVE) ([]rpm.Checker, error) {
+	var chks []rpm.Checker
+
+	for _, ar := range cve.AffectedRelease {
+		if ar.CPE == "" {
+			continue
+		}
+
+		d, err := wfn.Parse(ar.CPE)
+		if err != nil {
+			return nil, fmt.Errorf("can't parse distro cpe %q: %v", ar.CPE, err)
+		}
+		// TODO we need to do this because RedHat sometimes sets `a` as part for RHEL-X, when it should be `o`
+		d.Part = wfn.Any
+
+		var pc pkgCheck = constPkgChecker(true) // match all packages
+		if ar.Package != "" {
+			// add .src to parse it correctly, they're all src rpms
+			if p, err := rpm.Parse(ar.Package + ".src"); err == nil {
+				pc = affectedReleasePkgChecker(*p)
+			}
+		}
+
+		chks = append(chks, &singleChecker{d, pc})
+	}
+
+	return chks, nil
+}
+
+func packageStateCheckers(cve *schema.CVE) ([]rpm.Checker, error) {
+	var chks []rpm.Checker
+
+	for _, ps := range cve.PackageState {
+		if ps.FixState != "" && !schema.IsFixed(ps.FixState) {
+			// if the package hasn't been fixed, continue
+			continue
+		}
+
+		if ps.CPE == "" {
+			continue
+		}
+
+		d, err := wfn.Parse(ps.CPE)
+		if err != nil {
+			return nil, fmt.Errorf("can't parse distro cpe %q: %v", ps.CPE, err)
+		}
+		// TODO we need to do this because RedHat sometimes sets `a` as part for RHEL-X, when it should be `o`
```

Comment:
Why is it a TODO? I don't think there's any action pending for that. Maybe `XXX`, or just drop the tag?